### PR TITLE
Android theme revamp

### PR DESCRIPTION
**Pull request by Nikokoli**

### CODE REVIEW

File: app/src/main/java/fi/haltu/harrastuspassi/fragments/PromotionFragment.kt [hunk 1]
```diff
@@ -31,6 +32,7 @@ import org.json.JSONArray
 import org.json.JSONException
 import org.json.JSONObject
 import java.io.IOException
+import java.lang.reflect.Type
 import java.net.URL
```
Comment (Doatti) [inline app/src/main/java/fi/haltu/harrastuspassi/fragments/PromotionFragment.kt:38]: Imports added but no lines added that use them?

File: app/src/main/java/fi/haltu/harrastuspassi/fragments/home/HomeHobbiesFragment.kt [hunk 2]
```diff
@@ -24,7 +23,6 @@ import fi.haltu.harrastuspassi.utils.createHobbyEventQueryUrl
 import fi.haltu.harrastuspassi.utils.idToWeekDay
 import fi.haltu.harrastuspassi.utils.loadFilters
 import fi.haltu.harrastuspassi.utils.verifyAvailableNetwork
-import org.json.JSONArray
```
Comment (Doatti) [inline app/src/main/java/fi/haltu/harrastuspassi/fragments/home/HomeHobbiesFragment.kt:27]: Are you sure we don't need this? Does line 211 still work without this import?

File: app/src/main/res/layout/adapter_horizontal_hobby_list_item.xml [hunk 3]
```diff
@@ -1,79 +1,81 @@
...
+            android:layout_height="wrap_content"
+            android:layout_marginStart="8dp"
+            android:fontFamily="@font/poppins_regular"
+            android:textColor="@color/hobbyBlue"
+            android:textSize="14sp"
+            app:layout_constraintBottom_toBottomOf="@+id/horizontal_hobby_list_week_day_icon"
+            app:layout_constraintEnd_toEndOf="@+id/horizontal_hobby_list_description"
+            app:layout_constraintStart_toEndOf="@+id/horizontal_hobby_list_week_day_icon"
+            app:layout_constraintTop_toTopOf="@+id/horizontal_hobby_list_week_day_icon" />
     </androidx.constraintlayout.widget.ConstraintLayout>
```
Comment (Doatti) [inline app/src/main/res/layout/adapter_horizontal_hobby_list_item.xml:80]: Unnecessary whitespace changes and code re-organization. These should be in a separate PR

File: app/src/main/java/fi/haltu/harrastuspassi/fragments/home/HomeHobbiesFragment.kt (hunk 2 above)
Comment (Nikokoli) [reply app/src/main/java/fi/haltu/harrastuspassi/fragments/home/HomeHobbiesFragment.kt:27]: I didn't even notice deleting this, Line 211 refers to JSONObject, not JSONArray so yeah it's actually unused import

File: app/src/main/res/layout/adapter_horizontal_hobby_list_item.xml (hunk 3 above)
Comment (Nikokoli) [reply app/src/main/res/layout/adapter_horizontal_hobby_list_item.xml:80]: I guess these whitespace changes and code re-organisations are made automatically by my Android Studio config 🤯

File: app/src/main/java/fi/haltu/harrastuspassi/fragments/home/HomeHobbiesFragment.kt (hunk 2 above)
Comment (Doatti) [reply app/src/main/java/fi/haltu/harrastuspassi/fragments/home/HomeHobbiesFragment.kt:27]: I meant this line `val mJsonArray = results.getJSONArray("results")`. It doesn't refer to JSONArray explicitly but I guess that's the class of the object that gets returned from `.getJSONArray()`

Comment (Doatti) [reply app/src/main/java/fi/haltu/harrastuspassi/fragments/home/HomeHobbiesFragment.kt:27]: It's probably fine but it's always risky to do this kind of changes especially when it's clearly unrelated to what you are trying to do in this PR.
